### PR TITLE
[mono][debugger] Do not use `select(2)` on debugger-agent

### LINKED ISSUE #82493: Stop using select in the Mono debugger agent
As discussed in https://github.com/dotnet/runtime/issues/82428, avoid using the `select` routine in the Mono debugger agent to handle file descriptors larger than `FD_SETSIZE`.

`select` is currently used in`src/mono/mono/component/debugger-agent.c`.

### PR DESCRIPTION
**Pull request by thaystg**
Avoid using the select routine in the Mono debugger agent to handle file descriptors larger than FD_SETSIZE
Because of this PR: https://github.com/dotnet/runtime/pull/82429

Fixes #82493

### CODE REVIEW

**Comment by radical** [comment]
Could you please add the reasoning for not using `select` in the PR description?

**Comment by thaystg** [comment]
> Could you please add the reasoning for not using `select` in the PR description?

Done.

File: src/mono/mono/component/debugger-agent.c [hunk 1]
```diff
@@ -1144,16 +1145,10 @@ socket_transport_connect (const char *address)
...
-			tv.tv_usec = agent_config.timeout * 1000;
-			FD_ZERO (&readfds);
-			FD_SET (sfd, &readfds);
-
-			res = select ((int)sfd + 1, &readfds, NULL, NULL, &tv);
-
+			mono_pollfd mp;
+			mp.fd = (int)sfd;
+			mp.events = MONO_POLLIN;
+			res = mono_poll (&mp, 1, agent_config.timeout);
```
Comment (ilonatommy) [inline src/mono/mono/component/debugger-agent.c:1151]: We're not multiplying by 1k anymore? Inside of `mono_poll` we are dividing by 1k
https://github.com/dotnet/runtime/blob/3aec142ed1cb29e2a2f0588900347eee491b9c51/src/mono/mono/utils/mono-poll.c#L75
so maybe we should keep it upscaled?

**Review by ilonatommy — APPROVED** [review]

**Comment by thaystg** [comment]
@lambdageek can you please review this PR when you have time?

File: src/mono/mono/component/debugger-agent.c (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/component/debugger-agent.c:1151]: `agent_config.timeout` is in milliseconds.   `tv.tv_usec` was in microseconds.   `mono_poll` also takes a timeout in milliseconds.  There are actually several implementations.  The one where we're dividing is converting from milliseconds to seconds+microseconds.

tl;dr there's no need to do any conversion here at the call site for `mono_poll`

**Review by lambdageek — APPROVED** [review]
lgtm

**Comment by lambdageek** [comment]
Note that `mono_poll` is still using `select` on apple platforms. I'm not sure if that's a problem

**Comment by thaystg** [comment]
> Avoid using the select routine in the Mono debugger agent to handle file descriptors larger than FD_SETSIZE Because of this PR: #82429
> 
> Fixes #82493

mono_pool is not used anywhere, should I test it on macos and enable it for apple also?
